### PR TITLE
fix(ics27): check packet data length explicitly over nil check

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Use explicit `len()` check of interchain account packet data 

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by crodriguezvega** [comment]
Is this change state-machine breaking, @damiannolan?

**Comment by damiannolan** [comment]
> Is this change state-machine breaking, @damiannolan?

Yes, actually I think it probably is. Considering the check is in `ValidateBasic` and would previous allow empty `[]byte` to proceed then I think it would have state machine breaking effects, right? I can update the changelog. cc. @colin-axner

**Review by colin-axner — APPROVED** [review]
Thanks! We should backport to all next major/minor releases starting at v3

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +92,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (apps/29-fee) [\#1774](https://github.com/cosmos/ibc-go/pull/1774) Change non nil relayer assertion to non empty to avoid import/export issues for genesis upgrades. 
 * (makefile) [\#1785](https://github.com/cosmos/ibc-go/pull/1785) Fetch the correct versions of protocol buffers dependencies from tendermint, cosmos-sdk, and ics23.
 * (apps/29-fee) [\#1278](https://github.com/cosmos/ibc-go/pull/1278) The URI path for the query to get all incentivized packets for a specific channel did not follow the same format as the rest of queries.
+* (apps/27-interchain-accounts) [\#1882](https://github.com/cosmos/ibc-go/pull/1882) Explicitly check length of interchain account packet data in favour of nil check.
```
Comment (seantking) [inline CHANGELOG.md:95]: changelog should be moved to #State Machine Breaking section

**Review by seantking — APPROVED** [review]

**Pull request merged**